### PR TITLE
virt-launcher: fix enabled cpu num

### LINKED ISSUE #10462: VM with liveUpdateFeatures is created with only one socket
**What happened**:
I create VM with liveUpdateFeatures enabled feature and with the following cpu topology:

```yaml
spec:
  liveUpdateFeatures:
    cpu: {}
  template:
    spec:
      domain:
        cpu:
          cores: 1
          model: Nehalem
          threads: 1
          sockets: 3
```

After creating the VM, I only see 1 CPU (although I expected to see 3, according to topology).

**What you expected to happen**:
The number of CPUs in the running machine corresponds to the number specified in the CPU topology.

**How to reproduce it (as minimally and precisely as possible)**:
Just deploy this manifest:

```yaml
apiVersion: v1
kind: Namespace
metadata:
  labels:
    vm: vm101
  name: vms
---
apiVersion: cdi.kubevirt.io/v1beta1
kind: DataVolume
metadata:
  annotations:
    cdi.kubevirt.io/storage.bind.immediate.requested: "true"
    cdi.kubevirt.io/storage.deleteAfterCompletion: "false"
  labels:
    vm: vm101
  name: linux-boot-101
  namespace: vms
spec:
  pvc:
    accessModes:
    - ReadWriteMany
    resources:
      requests:
        storage: 4Gi
    volumeMode: Block
  source:
    registry:
      pullMethod: node
      url: docker://docker.io/fl64/alpine-3-18:latest
---
apiVersion: kubevirt.io/v1
kind: VirtualMachine
metadata:
  labels:
    vm: vm101
  name: vm-linux-101
  namespace: vms
spec:
  liveUpdateFeatures:
    cpu: {}
  running: true
  template:
    metadata:
      annotations:
        kubevirt.io/allow-pod-bridge-network-live-migration: ""
      labels:
        kubevirt.io/vm: vm-linux
        vm: vm101
    spec:
      domain:
        cpu:
          cores: 1
          model: Nehalem
          sockets: 3
          threads: 1
        devices:
          interfaces:
          - bridge: {}
            model: virtio
            name: default
        memory:
          guest: 1Gi
      networks:
      - name: default
        pod: {}
      terminationGracePeriodSeconds: 3
      volumes:
      - dataVolume:
          name: linux-boot-101
        name: boot
```

**Additional context**:
If I change the number of sockets, for example to 2, after the livemigration process I see 2 CPUs.

Everything fine with sockets count without liveUpdateFeatures. I see all 3 CPUs.

**Environment**:
- KubeVirt version (use `virtctl version`): v1.0.0
- Kubernetes version (use `kubectl version`): v1.26.8
- VM or VMI specifications: N/A
- Cloud provider or hardware configuration: Baremetal
- OS (e.g. from /etc/os-release): ubuntu 22.04
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others: N/A

### PR DESCRIPTION
**Pull request by fl64**
**What this PR does / why we need it**:
When creating a VM with `liveFeatureUpdate: { cpu: {} }`, the enabled vCpu calculation is incorrect because the number of sockets, which can be more than 1 when creating a VM, is not taken into count.

**Which issue(s) this PR fixes**  #
Fixes #10462

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by enp0s3** [comment]
@fl64 hey! thanks for taking care of this
/cc

**Review by 0xFelix — COMMENTED** [review]
/ok-to-test

Looks good to me, waiting what CI says

**Comment by enp0s3** [comment]
@0xFelix @fl64 it worth waiting for the [revert](https://github.com/kubevirt/kubevirt/pull/10480) to get in and then retest the comupte-migrations

File: pkg/virt-launcher/virtwrap/converter/converter_test.go [hunk 1]
```diff
@@ -828,20 +828,29 @@ var _ = Describe("Converter", func() {
 			It("should define hotplugable default topology", func() {
 				v1.SetObjectDefaults_VirtualMachineInstance(vmi)
 				vmi.Spec.Domain.CPU = &v1.CPU{
-					MaxSockets: 4,
+					Cores:      2,
+					MaxSockets: 3,
```
Comment (xpivarc) [inline pkg/virt-launcher/virtwrap/converter/converter_test.go:832]: Any reason to change these?

File: pkg/virt-launcher/virtwrap/converter/converter_test.go [hunk 2]
```diff
@@ -828,20 +828,29 @@ var _ = Describe("Converter", func() {
...
+				Expect(domainSpec.VCPUs.VCPU[2].Hotpluggable).To(Equal("yes"), "Expecting the 3rd vcpu to be Hotpluggable")
+				Expect(domainSpec.VCPUs.VCPU[3].Hotpluggable).To(Equal("yes"), "Expecting the 4th vcpu to be Hotpluggable")
+				Expect(domainSpec.VCPUs.VCPU[4].Hotpluggable).To(Equal("yes"), "Expecting the 5th vcpu to be Hotpluggable")
+				Expect(domainSpec.VCPUs.VCPU[5].Hotpluggable).To(Equal("yes"), "Expecting the 6th vcpu to be Hotpluggable")
+				Expect(domainSpec.VCPUs.VCPU[0].Enabled).To(Equal("yes"), "Expecting the 1st vcpu to be enabled")
+				Expect(domainSpec.VCPUs.VCPU[1].Enabled).To(Equal("yes"), "Expecting the 2nd vcpu to be enabled")
+				Expect(domainSpec.VCPUs.VCPU[2].Enabled).To(Equal("yes"), "Expecting the 3rd vcpu to be enabled")
+				Expect(domainSpec.VCPUs.VCPU[3].Enabled).To(Equal("yes"), "Expecting the 4th vcpu to be enabled")
+				Expect(domainSpec.VCPUs.VCPU[4].Enabled).To(Equal("no"), "Expecting the 5th vcpu to be disabled")
+				Expect(domainSpec.VCPUs.VCPU[5].Enabled).To(Equal("no"), "Expecting the 6th vcpu to be disabled")
```
Comment (xpivarc) [inline pkg/virt-launcher/virtwrap/converter/converter_test.go:853]: nit: Can we do `for` here?

**Review by xpivarc — COMMENTED** [review]
/approve

**Review by kvaps — APPROVED** [review]

**Pull request merged**